interfaces/acados_template/acados_template/acados_ocp_batch_solver.py

```diff
@@ -51,10 +52,22 @@ class AcadosOcpBatchSolver():
 
     __ocp_solvers : List[AcadosOcpSolver]
 
-    def __init__(self, ocp: AcadosOcp, N_batch: int, json_file: str = 'acados_ocp.json',  build: bool = True, generate: bool = True, verbose: bool=True):
+    def __init__(self, ocp: AcadosOcp, N_batch: int, num_threads_in_batch_solve: Union[int, None] = None, json_file: str = 'acados_ocp.json',  build: bool = True, generate: bool = True, verbose: bool=True):
 
         if not isinstance(N_batch, int) or N_batch <= 0:
             raise Exception("AcadosOcpBatchSolver: argument N_batch should be a positive integer.")
+        if num_threads_in_batch_solve is None:
+            num_threads_in_batch_solve = ocp.solver_options.num_threads_in_batch_solve
+            print(f"Warning: num_threads_in_batch_solve is None. Using value {num_threads_in_batch_solve} set in ocp.solver_options instead.")
+            print("In the future, it should be passed explicitly in the AcadosOcpBatchSolver constructor.")
```

Comment:
It would be good to also put the deprecation warning in the setter of `num_threads_in_batch_solve`.
Also, since `num_threads_in_batch_solve` is not used in the template logic anymore, and as it was never functional in MATLAB, we can remove the option in all .m files.